### ISSUE TITLE
testing: add more complete unit tests for `type_traits` and `expressions`

### ISSUE THREAD

**Issue by fnrizzi** [Pressio/pressio#302]
@MikolajZuzek can you please help me out with this? 

As I am making progress with the refactor, it would be great if you follow along with tests. 
Right now, I am mostly done with type_traits and expressions. 
Please use this branch: `refactor_issue298`

- `type_traits`:
  - code: https://github.com/Pressio/pressio/tree/refactor_issue298/include/type_traits
  - unit tests should be added here:  https://github.com/Pressio/pressio/tree/refactor_issue298/unit_tests/type_traits
 
- `expressions`: 
  - code: https://github.com/Pressio/pressio/tree/refactor_issue298/include/expressions
  - unit tests should be added here:  https://github.com/Pressio/pressio/tree/refactor_issue298/unit_tests/expressions

**Comment by MikolajZuzek** [comment]
@fnrizzi  What's the idea behind `kokkos_strided_subview` test in [kokkos_strided_subview.cc](https://github.com/Pressio/pressio/blob/refactor_issue298/tests/functional_small/expressions/kokkos_strided_subview.cc) ?  
It doesn't test anything from Pressio... Is it to be expanded or something ?

**Comment by fnrizzi** [comment]
no, just there to test the native kokkos functionality :)

**Comment by fnrizzi** [comment]
@MikolajZuzek i don't want to get rid of it though , good to have for reference

**Comment by MikolajZuzek** [comment]
@fnrizzi We could separate TPL tests into a different folder.

**Comment by fnrizzi** [comment]
Yes I agree, for now let's not worry about that, I like to prioritize the actual tests and we can reorganize later
